src/dpo/src/detailed.cxx

```diff
@@ -0,0 +1,229 @@
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <algorithm>
+#include <boost/format.hpp>
+#include <boost/tokenizer.hpp>
+#include <cmath>
+#include <iostream>
+#include <stack>
+#include <utility>
+#include "utl/Logger.h"
+#include "plotgnu.h"
+#include "utility.h"
+
+// Detailed management of segments.
+#include "detailed_manager.h"
+#include "detailed_segment.h"
+// Detailed placement algorithms.
+#include "detailed.h"
+#include "detailed_global.h"
+#include "detailed_interleaving.h"
+#include "detailed_mis.h"
+#include "detailed_random.h"
+#include "detailed_reorder.h"
+#include "detailed_vertical.h"
+// Other.
+#include "detailed_orient.h"
+
+// Other things not ready.
+//#include "detailed_global_vertical.h"
+// Detailed placement objectives.
+//#include "detailed_hpwl.h"
+//#include "detailed_drc.h"
+//#include "detailed_abu.h"
+//#include "detailed_pin.h"
+
+using utl::DPO;
+
+namespace dpo {
+
+////////////////////////////////////////////////////////////////////////////////
+// Defines.
+////////////////////////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////////////////////////
+// Detailed::~Detailed:
+////////////////////////////////////////////////////////////////////////////////
+Detailed::~Detailed(void) {}
+
+////////////////////////////////////////////////////////////////////////////////
+// Detailed::improve:
+////////////////////////////////////////////////////////////////////////////////
+bool Detailed::improve(DetailedMgr& mgr)
+// bool Detailed::improve( Architecture* arch, Network* network, RoutingParams*
+// rt )
+{
+  m_mgr = &mgr;
+
+  m_arch = mgr.getArchitecture();
+  m_network = mgr.getNetwork();
+  m_rt = mgr.getRoutingParams();  // Can be NULL.
+
+
+  int err1, err2, err3, err4, err5;
+  (void)err1;
+  (void)err2;
+  (void)err3;
+  (void)err4;
+  (void)err5;
+
+  // Checks prior to detailed placement.
+  err1 = mgr.checkRegionAssignment();  // If segs build right and assignment
+                                       // okay, should be fine.
+  err2 = mgr.checkRowAlignment();   // If segs build right and assignment okay,
+                                    // should be fine.
+  err3 = mgr.checkSiteAlignment();  // Might not be okay if initial placement
+                                    // not legal or code made a mistake.
+  err4 =
+      mgr.checkOverlapInSegments(10);  // Might not be okay if initial placement
```

Comment:
We should refrain from using magic numbers that don't have context. We should give this value a variable name and explain its value.